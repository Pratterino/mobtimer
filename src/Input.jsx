import React, {Component} from 'react';
import PropTypes from 'prop-types';
import './Input.scss';

class Input extends Component {
    constructor(props) {
        super(props);
        this.state = {
            text: "",
        };
    }

    handleSubmit = (e) => {
        e.preventDefault();
        this.props.handleInputSubmit(this.state.text);
        this.setState({
            text: "",
        });
    };

    render() {
        return (
            <form className="Input" onSubmit={this.handleSubmit}>
                <div className="input-container">
                    <div className="parent">
                        <label
                            className="add"
                            htmlFor="input-name"
                        >Add name</label>
                        <input
                            name="input-name"
                            value={this.state.text}
                            className="message"
                            autoComplete="off"
                            type="text"
                            onChange={(e) => {
                                this.setState({text: e.target.value})
                            }}
                        />
                    </div>
                </div>
            </form>
        );
    }
}

Input.propTypes = {
    handleInputSubmit: PropTypes.func.isRequired,
};
export default Input;

