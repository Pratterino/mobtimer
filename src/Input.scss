@import url('https://fonts.googleapis.com/css?family=Raleway:300,400,500');
@import url('https://fonts.googleapis.com/css?family=Inconsolata');

.Input {
    $width: 180px;
    $height: 40px;

    * {
        margin: 0;
        padding: 0;
    }

    .input-container {
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
    }

    label {
        position: absolute;
        left: 0;
        top: -24px;
        color: var(--header-color);
    }

    .parent {
        width: $width;
        height: $height;
        position: relative;

        .message {
            position: absolute;
        }
    }

    label,
    input.message {
        color: var(--header-color);
        border-color: var(--header-color);
        transition: 0.2s;
        &:focus {
            background: var(--background);
        }
    }

    .message {
        left: 0;
        width: inherit;
        background: none;
        border: 2px solid;
        font-size: 30px;
        padding-left: 10px;
    }
}
