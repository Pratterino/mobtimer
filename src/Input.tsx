import React, { useState } from 'react';
import './Input.scss';

interface IProps {
    handleInputSubmit: Function;
}

function Input({ handleInputSubmit }: IProps) {
    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        handleInputSubmit(inputValue);
        setInputValue('');
    };

    return (
        <form className="Input" onSubmit={handleSubmit}>
            <div className="input-container">
                <div className="parent">
                    <label htmlFor="input-name">Add user</label>
                    <input
                        name="input-name"
                        value={inputValue}
                        className="message"
                        autoComplete="off"
                        type="text"
                        onChange={e => {
                            setInputValue(e.target.value);
                        }}
                    />
                </div>
            </div>
        </form>
    );
}

export default Input;
