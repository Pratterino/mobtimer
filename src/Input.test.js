import React from 'react';
import Input from './Input';
import {shallow} from 'enzyme';

describe("Input", () => {
    const handleInputSubmit = jest.fn();
    let wrapper;

    const renderComponent = (props) => {
        wrapper = shallow(
            <Input
                handleInputSubmit={handleInputSubmit}
                {...props}
            />
        );
    };

    beforeEach(() => {
        renderComponent();
    });

    describe('handleSubmit', () => {
        it('should call handleInputSubmit', () => {
            let mockFunction = jest.fn();
            let e = {preventDefault: mockFunction};

            wrapper.instance().handleSubmit(e);

            expect(mockFunction).toHaveBeenCalledTimes(1);
            expect(handleInputSubmit).toHaveBeenCalledTimes(1);
            expect(wrapper.instance().state).toEqual({text: ""});
        });
    });

    describe('render', () => {
        it('should match snapshot', () => {
            //expect(wrapper).toMatchSnapshot();
        });
    });
});
