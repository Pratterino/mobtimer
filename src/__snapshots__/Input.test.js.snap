// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    handleInputSubmit={
      [MockFunction] {
        "calls": Array [
          Array [
            "test",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="input-container"
      >
        <div
          className="parent"
        >
          <label
            className="add"
            htmlFor="input-name"
          >
            Add user
          </label>
          <input
            autoComplete="off"
            className="message"
            name="input-name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>,
      "className": "Input",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="parent"
        >
          <label
            className="add"
            htmlFor="input-name"
          >
            Add user
          </label>
          <input
            autoComplete="off"
            className="message"
            name="input-name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>,
        "className": "input-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="add"
              htmlFor="input-name"
            >
              Add user
            </label>,
            <input
              autoComplete="off"
              className="message"
              name="input-name"
              onChange={[Function]}
              type="text"
              value=""
            />,
          ],
          "className": "parent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add user",
              "className": "add",
              "htmlFor": "input-name",
            },
            "ref": null,
            "rendered": "Add user",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "autoComplete": "off",
              "className": "message",
              "name": "input-name",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="input-container"
        >
          <div
            className="parent"
          >
            <label
              className="add"
              htmlFor="input-name"
            >
              Add user
            </label>
            <input
              autoComplete="off"
              className="message"
              name="input-name"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>,
        "className": "Input",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="parent"
          >
            <label
              className="add"
              htmlFor="input-name"
            >
              Add user
            </label>
            <input
              autoComplete="off"
              className="message"
              name="input-name"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>,
          "className": "input-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="add"
                htmlFor="input-name"
              >
                Add user
              </label>,
              <input
                autoComplete="off"
                className="message"
                name="input-name"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
            "className": "parent",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add user",
                "className": "add",
                "htmlFor": "input-name",
              },
              "ref": null,
              "rendered": "Add user",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "className": "message",
                "name": "input-name",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
